run_uh.py

```diff
@@ -294,53 +295,50 @@ def import_fife(paths):
 		return False
 	return True
 
-def find_fife():
+def get_fife_paths():
+	"""Returns all possible paths to search for the fife module. New paths to be added as needed."""
 	# Use the path the user provided.
 	from horizons.util.cmdlineoptions import get_option_parser
 	options = get_option_parser().parse_args()[0]
+	paths = []
+	# Check if path was provided by user.
 	if options.fife_path:
 		fife_path = os.path.abspath(options.fife_path)
+		paths.append(fife_path)
 		# Support giving the path to FIFE_ROOT/engine/python/fife/__init__.pyc etc.
 		if os.path.isfile(fife_path):
+			# Support giving the path to FIFE_ROOT/engine/python
 			fife_path = os.path.dirname(fife_path)
-		# Support giving the path to FIFE_ROOT/engine/python
-		if import_fife([fife_path]):
-			return True
-		# Support giving the path to FIFE_ROOT/engine
-		if import_fife([os.path.join(fife_path, 'python')]):
-			return True
-		# Support giving the path to FIFE_ROOT
-		if import_fife([os.path.join(fife_path, 'engine', 'python')]):
-			return True
-		# Support giving the path to FIFE_ROOT/engine/python/fife
-		if import_fife([os.path.join(fife_path, '..')]):
-			return True
-
-		# End the search to avoid using the wrong (non-user-specified) FIFE.
-		log().error('Unable to find FIFE in %s', fife_path)
-		exit(1)
-
-	# Try to use the default FIFE (equivalent of just trying to import it).
-	if import_fife(None):
-		return True
-
-	# Look for FIFE in the neighborhood of the game dir.
-	paths = []
+			paths.append(fife_path)
+			# Support giving the path to FIFE_ROOT/engine
+			paths.append(os.path.join(fife_path, 'python'))
+			# Support giving the path to FIFE_ROOT
+			paths.append(os.path.join(fife_path, 'engine', 'python'))
+			# Support giving the path to FIFE_ROOT/engine/python/fife
+			paths.append(os.path.join(fife_path, '..'))
+
+	# Look for FIFE in the neighborhood of the game dir:
 	for opt1 in ('.', '..', '..' + os.sep + '..'):
 		for opt2 in ('.', 'fife', 'FIFE', 'Fife', 'fifengine'):
 			for opt3 in ('.', 'trunk'):
 				path = os.path.abspath(os.path.join('.', opt1, opt2, opt3, 'engine', 'python'))
 				if os.path.exists(path):
 					paths.append(path)
-	return import_fife(paths)
+
+	return paths
 
 def setup_fife():
 	log_paths()
 	log_sys_info()
-	if not find_fife():
-		#TODO useful error message anyone?
-		exit_with_error('Failed to find and/or load FIFE', 'Make sure FIFE was installed as a package/redistributable or compiled from source.')
+	paths = get_fife_paths()
+	if not find_fife(paths):
+		try:
+			from fife import fife
+		except ImportError:
+			directories = '\n'.join(paths)
+			exit_with_error('Failed to load module fife', 'Below directory paths were tested:\n' + directories)
 
+	# Another import call still needed?
```

Comment:
I think it is needed. We do import fife before (line 336), but that code isn't always executed. The only other import happens in `find_fife` which isn't run in the global scope.